pandas/indexes/multi.py

```diff
@@ -2069,20 +2069,14 @@ def convert_indexer(start, stop, step, indexer=indexer, labels=labels):
         else:
 
             loc = level_index.get_loc(key)
-            if level > 0 or self.lexsort_depth == 0:
+            if isinstance(loc, slice):
+                return loc
+            elif level > 0 or self.lexsort_depth == 0:
                 return np.array(labels == loc, dtype=bool)
-            else:
-                # sorted, so can return slice object -> view
-                try:
-                    loc = labels.dtype.type(loc)
```

Comment:
WOW!
```
In [4]: arr = np.arange(10000000, dtype='int32')

In [5]: %timeit arr.searchsorted(np.int32(9999999))
The slowest run took 17.48 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 1.5 µs per loop

In [6]: %timeit arr.searchsorted(9999999)
10 loops, best of 3: 24.2 ms per loop
```